src/algebra/module/graded_module.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import ring_theory.graded_algebra.basic
+import algebra.direct_sum.decomposition
+
+/-!
+# Graded Module
+
+Given an `R`-algebra `A` graded by `𝓐`, a graded `A`-module `M` is expressed as
+`direct_sum.decomposition 𝓜` and `set_like.has_graded_scalar 𝓐 𝓜`.
+Then `⨁ i, 𝓜 i` is an `A`-module and is isomorphic to `M`.
+
+## Tags
+
+graded module
+-/
+
+
+open_locale direct_sum big_operators
+
+variables {ι R A M σ σ' : Type*}
+variables [add_monoid ι] [comm_semiring R] [semiring A] [algebra R A]
+variables (𝓐 : ι → σ') [set_like σ' A]
+variables (𝓜 : ι → σ)
+
+namespace graded_module
+
+include σ' A
+
+section self
+
+variable [add_submonoid_class σ' A]
+
+/--each graded ring is also a graded module over itself.-/
+lemma graded_algebra.to_graded_module [decidable_eq ι]
+  [graded_ring 𝓐] :
+  set_like.has_graded_scalar 𝓐 (λ i, (𝓐 i)) :=
+{ smul_mem := λ i j x y hi hj, set_like.graded_monoid.mul_mem hi hj }
```

Comment:
This looks like it should be called `set_like.graded_monoid.to_has_graded_scalar` to me, and belongs in the other file.